wsd/QuarantineUtil.cpp

```diff
@@ -90,6 +132,8 @@ namespace Quarantine
 
         std::sort(files.begin(), files.end());
 
+        deleteExpiredFiles(files);
+
         std::size_t currentSize = quarantineSize();
         std::size_t index = 0;
         while (currentSize >= sizeLimit && !files.empty())
```

Comment:
So doing a single state in this loop below; and having some nice intermediates to flag what we are detecting and doing some easy-to-read logic around that ?

bool isExpired = <timestamp-check>
bool isInUse = hardlink-count > 1

if (isExpired || !inUse || totalSize > whatever) ...
   erase.